app_dart/lib/src/service/luci_build_service.dart

```diff
@@ -224,27 +265,131 @@ class LuciBuildService {
     // Ensure we are using V2 bucket name istead of V1.
     // V1 bucket name  is "luci.flutter.prod" while the api
     // is expecting just the last part after "."(prod).
-    final String bucketName = build.bucket.split('.').last;
+    final String bucketName = buildPushMessage.build.bucket.split('.').last;
+    final Map<String, dynamic> userData =
+        jsonDecode(buildPushMessage.userData) as Map<String, dynamic>;
+    userData['retries'] += 1;
     await buildBucketClient.scheduleBuild(ScheduleBuildRequest(
       builderId: BuilderId(
-        project: build.project,
+        project: buildPushMessage.build.project,
         bucket: bucketName,
         builder: builderName,
       ),
       tags: <String, List<String>>{
-        'buildset': build.tagsByName('buildset'),
-        'user_agent': build.tagsByName('user_agent'),
-        'github_link': build.tagsByName('github_link'),
+        'buildset': buildPushMessage.build.tagsByName('buildset'),
+        'user_agent': buildPushMessage.build.tagsByName('user_agent'),
+        'github_link': buildPushMessage.build.tagsByName('github_link'),
       },
-      properties: (build.buildParameters['properties'] as Map<String, dynamic>)
+      properties: (buildPushMessage.build.buildParameters['properties']
+              as Map<String, dynamic>)
           .cast<String, String>(),
       notify: NotificationConfig(
         pubsubTopic: 'projects/flutter-dashboard/topics/luci-builds',
-        userData: json.encode(<String, dynamic>{
-          'retries': retries + 1,
-        }),
+        userData: json.encode(userData),
       ),
     ));
     return true;
   }
+
+  /// Sends a [BuildBucket.scheduleBuild] request using [CheckRunEvent]. It
+  /// returns [true] if it is able to send the scheduleBuildRequest or [false]
+  /// if not.
+  Future<bool> rescheduleUsingCheckRunEvent(CheckRunEvent checkRunEvent) async {
+    final github.RepositorySlug slug = checkRunEvent.repository.slug();
+    final Map<String, dynamic> userData = <String, dynamic>{};
+    final github.PullRequest pr = checkRunEvent.checkRun.pullRequests[0];
+    final github.GitHub gitHubClient =
+        await config.createGitHubClient(slug.owner, slug.name);
+    final github.CheckRun githubCheckRun =
+        await githubChecksUtil.createCheckRun(
+      gitHubClient,
+      slug,
+      checkRunEvent.checkRun.name,
+      pr.head.sha,
+    );
+    userData['check_suite_id'] = checkRunEvent.checkRun.checkSuite.id;
+    userData['check_run_id'] = githubCheckRun.id;
+    userData['repo_owner'] = slug.owner;
+    userData['repo_name'] = slug.name;
+    userData['user_agent'] = 'flutter-cocoon';
+    userData['retries'] = 1;
```

Comment:
It is a little bit confusing with initialization 1 in a rescheduleXXX function without any doc explanation.

Is this the initialization value for any new run or any reschedule? If yes, the following `retries` is meant for re-runs of any new run or any reschedule run?

People may easily confuse the reschedule with retries. Some doc. would be helpful.

Reply:
Added docs. Actually retries will be removed as devs will be able to rerun builds manually if something fails.